### ISSUE TITLE
Document deployment for different technologies (Apache, GitHub pages, Netlify)

### ISSUE THREAD

**Issue by BoilerTom90** [magidoc-org/magidoc#145]
I'm pretty happy with the resulting documentation that's generated, and I can nicely preview it using the magidoc preview command.
However, I'm struggling to view the results outside of the preview command.

I tried opening the index.html with chrome and it errors out as soon as it can't find the "Introduction" page. 

I then experimented with the `siteRoot` option, and I feel like I got a little farther, but I still can't view the results outside of the `preview` command.  Each time the browser fails with a 404 when trying to redirect to the Introduction page.

I even tried installing apache from xampp and trying to serve up the web pages with it. Same result. It fails when trying to redirect to the Introduction page.  That is, I copied the entire build output into a directory named /standalone-server inside of apache's htdocs directory.   It too got a 404 trying to fetch the introduction page. 

My index.html file looks like this:

```html
<meta http-equiv="refresh" content="0;url=/standalone-server/introduction">
```
The introduction.html file is indeed in the /standalone-server directory. 

A few of my options are as follows:

```js
      options: {       
         siteRoot: '/standalone-server',
         appLogo: '/standalone-server/shure-logo.png',
         appFavicon: '/standalone-server/shure-favicon.png',
         ... 
     }
```

My generated output looks like this (excluding the mutations, queries, subscriptions, types, and app directory)

📦standalone-server
 ┣ 📜.nojekyll
 ┣ 📜CNAME
 ┣ 📜google1446b24f939c3f7a.html
 ┣ 📜index.html
 ┣ 📜introduction.html
 ┣ 📜manifest.json
 ┣ 📜robot.txt
 ┣ 📜shure-favicon.png
 ┣ 📜shure-logo.png
 ┗ 📜shure-logo2.png

**Comment by pelletier197** [comment]
> Your file output made me realise there was an issue with the script that publishes the starter. You have some assets in there that should not be present. Thanks for that!

# About siteRoot
`siteRoot` property is meant to be used if you want to host your website on a non-root URL. Take for example that your company's website is `https://acme.com` and you want to have the root of your documentation to be at `https://acme.com/docs`, then you would need to set `siteRoot` property to `/docs`. This means that all the content inside your website would start at this URL (`/docs/introduction`, `/docs/types/String`, etc).

# Deployment
Magidoc generates static HTML files, which means that any web-server can deliver the content. If you want to deploy it locally, you could use [serve](https://www.npmjs.com/package/serve) for instance
- `npx serve` (while using `siteRoot: '/standalone-server'`), which would put your documentation at `/standalone-server/introduction` when opened in the browser. 
- or `cd standalone-server && npx serve` (while leaving `siteRoot` undefined), which would leave your documentation at `/introduction` if opened in the browser.

If you use an http-server like Apache, you need to put your build inside the target folder of the html content (see [documentation](https://httpd.apache.org/docs/trunk/da/urlmapping.html#documentroot)), which may be the following location for instance `/var/www/html/index.html`, `/var/www/html/introduction.html`, etc (considering your `siteRoot` is undefined).

Since magidoc uses clean URLs (`/introduction` rather than `/introduction.html`), your HTTP server must be configured to use the extension `.html` by default if none is specified, which is the default for sirv and many other implementations, but **not for Apache** (which may explain your bug). I'm not really an Apache expert, but I think you should get around using [MultiViews](https://www.youtube.com/watch?v=Tp-WacMe7yA).

Hope this help, let me know if you figure out how to do it in your case. I would like to update the documentation to add a `deployment` section.

**Comment by BoilerTom90** [comment]
Hi, After I created the issue on github, I continued to look into this. I
mucked with the site root and figured out how that is intended to be used.
It makes sense now.

Also, I discovered the web servers I was using to try to view the pages
(e.g the live server vscode extension and other similar ones), didn't
handle the CLEAN URLs.

I had a local version of Apache (from an xampp installation) on my windows
machine. I created a rewrite rule to append .HTML to the URL if the URL
wasn't a directory path.  With the rewrite rule, xampp Apache was able to
serve up the site perfectly.

I'll try to MultiView solution as that's a lot simpler to setup.

Glad my file/dir output helped spot an issue with the extra assets.  I'll
keep an eye out for an updated release.

Tom


*~~~~~~~~~~~~~~~~~~~*
*Tom Hoffman*
